### ISSUE TITLE
License does not have xray product

### ISSUE THREAD

**Issue by EdmundDXu** [Lama3L9R/ArtifactoryKeygen#5]
Reproduce procedure:
`
$ curl -u admin:password http://localhost:8082/artifactory/api/xray/license

{
  "errors" : [ {
    "status" : 400,
    "message" : "{\"error\":\"License does not have xray product\"}"
  } ]
}
`

Seems like Artifactory 7.90.9 with ArtifactoryAgent-1.0-790fix-SNAPSHOT-all.jar agent does not support Xray。
This is my license：
![image](https://github.com/user-attachments/assets/f7bcd183-b3b5-4595-9ac1-2931dfa83ba6)
 
Based on the official page of JFrog Xray, this license should include Xray: 
![image](https://github.com/user-attachments/assets/e74e3332-5989-49b2-a40e-cf442eaa5a7e)

Did I miss anything?

**Comment by lamadaemon** [comment]
Interesting discovery! I have never tried X-ray before. Perhaps the way JFrog did was when they sell you Enterprise+ Licenses, the license you received is actually two licenses. There can be multiple products in one license. My keygen should supports adding multiple licenses. Try to add xray or x-ray as the product id (because I am not sure about the license ID) using the keygen. If this doesn't work, I will try to do some research and find out what's going on.

**Comment by EdmundDXu** [comment]
After adding the xray product id to the keygen, the Artifactory API `/artifactory/api/xray/license` no longer reports an error. 

`
$ curl -u admin:password http://localhost:8082/artifactory/api/xray/license

cHJvZHVjdHM6CiAgeHJheToKICAgIHByb2R1Y3Q6IFpYaHdhWEpsY3pvZ01qRXdNQzB3TVMwek1W
UXdNem96T1Rvek9TNDRNekphQ21sa09pQjRjbUY1Q205M2JtVnlPaUJzWVcxaFpHRmxiVzl1Q25C
eWIzQmxjblJwWlhNNklIdDlDbk5wWjI1aGRIVnlaVG9nYm5Wc2JBcDBjbWxoYkRvZ1ptRnNjMlVL
ZEhsd1pUb2dSVTVVUlZKUVVrbFRSVjlRVEZWVENuWmhiR2xrUm5KdmJUb2dNakF5TkMwd09TMHhO
RlF3TXpvek9UbzBNQzQyT0RCYUNnPT0KICAgIHNpZ25hdHVyZTogSkQxd1BtZHFjQjdhSEpqU1gv
YWFkZmo1REZCbytzRmtieXlITmc3VzZwUlpFaWVjMDRzL241MDJYZkYzenJkdWtsa2dNV083TEVK
ODBHeEtSOTgzWDBlUVhwSHZIL3pEaUM1NjA5RjI5OWhoc3RkOWJYWVVOS0R6N0ZqZFgxVTVjSy9H
eUVrNS9pMW1vSVE2bmh1NWs1VXh2RCt4WGdacVNZUXA0ekN6cnhkRnhBZ0hhT2RhMExHQ2tkQ3Fu
dGxiY1QrRkNvbXd5Ylh5Z1JlU0Iva3JlaEJCbWRNdFlYeTdLeWNzaDRSU05QcmUzZGUzQTBxZTZV
N1dkdk9PQi80SW1yaGFubU8xMUNlTi9MTTZEUHRMcWxZd2dkeEFzZEUydFZtS2hPZzVCTjRUODhD
cTJLRHdpbmZ4VnpKdjNyNVA0NDFXWlpPQUJGNkM2UlkrYnpEdmxyVXBxRzRIMVJPZDNESFpHLzdk
R0l0alYyRjF4SURDZHpNYi8vVDNoaVIxOW5qOXZwZTFDUm1LbTN5ck9EZnFNaXlmQXRkNEhiVzFm
eE9OTEl2S3VFR3BQN01rM3gzb0RINlBXc2dMT3EvRUFOVHIya0JyWGtXL3czVmZmS2Y0aUlSeXZq
ZW5aUHoweEpRelFudXQzanVjSFoydUo2RFRxMG9PYTNJaEpYb21IUnBoa2FvTzM0MnVja3JPRHpX
ekJ3eWRQNFpPODQyMzV6ai9ITWttaHppQmlwRjNsL1V5SWRJelNuUXZ0RmkrQWtDbzR6SHRtb2xZ
b0FCZ211VTE0SmY2S0Y0VTBZOG91clNENkllS043VGp0eUp0Q0N4enVLUE5oWHZWNW1scE9EVE5a
OWZoSTBGb1lzWWMvMXBtaW0waU5FMkdDZFdxOTV6V082aVkvNWs9CnZlcnNpb246IDEK
`

However, Xray is still showing the error 'License validation failed, Xray license is invalid.' in xray's `console.log`
![image](https://github.com/user-attachments/assets/7fadebf0-fbdb-4309-a040-791ca75da116)

**Comment by bykvaadm** [comment]
hello, @lamadaemon any updates? =)

**Comment by Lama3L9R** [comment]
Yes, an update is coming very soon. I was working on the new keygen these days and the version will be available on my website instead of a bad coded huge java application.

**Comment by bykvaadm** [comment]
Thanks for reply! will you update description in repo for new location of keygen? or where we can check for news of release of new keygen?

**Comment by Lama3L9R** [comment]
Recent update:

I think I know why X-ray won't work. It is because the agent isn't patching X-ray, and this requires further investigation. However, as I mentioned in another post that I am unable to deploy Artifactory on my local device, I won't publish the new keygen for now because it literally does the same thing except that you can use it online

**Comment by Lama3L9R** [comment]
I just downloaded Xray and have noticed Xray is a native application, which means ArtifactoryAgent will not work on it. Currently, I am doing reverse engineering on another native app, and I have no plan to work on Xray (my to-do list is full). However, keygen works. As long as you managed to replace the public key in Xray, the generated license should work.

**Comment by bykvaadm** [comment]
> As long as you managed to replace the public key in Xray

Could u please point a path in xray container where we should put public key?

**Comment by Lama3L9R** [comment]
key should be embedded in the app itself. Based on my experience, patch the license checking code will be easier than replace a constant.

**Comment by bykvaadm** [comment]
> > have noticed Xray is a native application
> 
> Hello, thanks for our work! May be do you know this binary name or parh to it. I could try to reverse...

Crack is nothing without signature databases, which is got by syncing with jfrog api, with authentication of course. We have successfully cracked xray, but it is useless piece of shit without repeatedly db sync
